server/container_create.go

```diff
@@ -258,6 +262,25 @@ func setupContainerUser(ctx context.Context, specgen *generate.Generator, rootfs
 		}
 	}
 
+	if genGroup && sc.RunAsGroup != nil {
```

Comment:
I think we should use gid here, assuming that we can take RunAsGroup into account for gid https://github.com/cri-o/cri-o/pull/8251/files#r1625011421

Reply:
Will make this change by setting gid to pick sc.RunAsGroup value (when not null)